enhancements/ingress/wildcard-routing.md

```diff
@@ -0,0 +1,189 @@
+---
+title: wildcard-routing-policy
+authors:
+  - "@danehans"
+reviewers:
+  - "@openshift/api-reviewers"
+  - "@openshift/sig-network-edge"
+  - "@danehans"
+
+approvers:
+  - "@openshift/api-approvers"
+creation-date: 2020-03-19
+last-updated: 2020-03-19
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Wildcard Routing Policy
+
+Application developers would like to expose a service outside the cluster by multiple names without specify each one
+individually.
+
+In OpenShift version 4, wildcard routing is not supported. The `Route` resource supports setting a wildcard policy but
+no API exists to enable this feature on an ingress controller (i.e. router).
+
+OpenShift version 3 supports this feature through the `ROUTER_ALLOW_WILDCARD_ROUTES` environment variable when
+configuring the OpenShift Router directly. Users of this feature are blocked from upgrading to OpenShift 4 unless they
+discontinue use of the feature.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+See the [section][#wildcard-routing-policy]
+
+## Motivation
+
+See the [section][#wildcard-routing-policy]
+
+### Goals
+
+1. Define an API field that allows a cluster administrator to enable wildcard routing for an ingress controller.
+
+### Non-Goals
+
+1. Redesign how wildcard routing is implemented in [OpenShift Router](https://github.com/openshift/router).
+2. Introduce a separate API for managing wildcard routing.
```

Comment:
Separate from what?

Reply:
Instead of adding a `WildcardPolicy` field to the existing `RouteAdmissionPolicy` API type, a non-goal is adding a separate API such as `WildcardAdmissionPolicy` to meet the use case. I'll spell this out more clearly in the EP.